ipa-core/src/protocol/ipa_prf/malicious_security/verifier.rs

```diff
@@ -21,81 +28,82 @@ pub struct ProofVerifier<F: PrimeField> {
 /// `https://eprint.iacr.org/2023/909.pdf`
 ///
 #[allow(non_camel_case_types)]
-impl<F> ProofVerifier<F>
+impl<F, λ> ProofVerifier<F, λ>
 where
     F: PrimeField,
+    λ: ArrayLength,
 {
-    pub fn new(u_or_v: Vec<F>, out_share: F) -> Self {
+    pub fn new(u_or_v: Vec<GenericArray<F, λ>>, out_share: F) -> Self {
         Self { u_or_v, out_share }
     }
 
-    pub fn verify_proof<λ>(
-        &self,
+    pub fn verify_proof<J, B>(
+        mut u_or_v_iterator: J,
```

Comment:
There's something strange about this argument being mutable. I wonder if there is any alternative.